packages/apple-pay-integration/src/apple-pay-customer-strategy.ts

```diff
@@ -543,29 +543,45 @@ export default class ApplePayCustomerStrategy implements CustomerStrategy {
     }
 
     private async _getBraintreeDeviceData() {
-        const data = await this._braintreeIntegrationService.getDataCollector();
+        let braintreePaymentMethod: PaymentMethod;
 
-        return data.deviceData;
+        try {
+            braintreePaymentMethod = this._paymentIntegrationService
+                .getState()
+                .getPaymentMethodOrThrow(ApplePayGatewayType.BRAINTREE);
+        } catch (_) {
+            return;
+        }
```

Comment:
It is not necessary to use try catch here. You can just remove orThrow.

```suggestion
            const braintreePaymentMethod = this._paymentIntegrationService
                .getState()
                .getPaymentMethod(ApplePayGatewayType.BRAINTREE);
```